Online content view tweaks and various bug fixes

samuelthomas2774 opened the pull request:
- Update online plugin/theme view styles
- Fix dropdowns
- Hide the edit button for packed plugins and themes
- Fix plugins not starting when refreshing content
- Change some Vue components to use events instead of callbacks
- Update inject script for new release format
- Move the client log file outside the client directory
- Fix reloading packed content
- Allow multiple users to use BetterDiscord on macOS and Linux by allowing paths relative to the user's home directory and allowing an additional per user config
- Stop watching theme files after they're unloaded
- Move updates to the correct location if the user changed it
- Fix native open dialogs (closes #338)

![Screenshot 2019-03-17 at 09 51 01](https://user-images.githubusercontent.com/8474065/54488513-5f25ff80-489a-11e9-95fe-92c5a3b3484c.png)

samuelthomas2774 committed:
Fix dropdowns

samuelthomas2774 committed:
Update online plugin/theme views to match local content views

samuelthomas2774 committed:
Hide the edit button for packed plugins/themes

samuelthomas2774 committed:
Vue events

samuelthomas2774 committed:
Merge remote-tracking branch 'upstream/master' into ui

samuelthomas2774 committed:
Remove debug logs

samuelthomas2774 committed:
Fix plugins not starting when manually loading all content

samuelthomas2774 committed:
Update inject script for new release format

samuelthomas2774 committed:
Use correct path to sparkplug

samuelthomas2774 committed:
Move the client log file outside the client directory

samuelthomas2774 committed:
Fix using a custom path to Discord.app on macOS

samuelthomas2774 committed:
Fix development build on Windows

samuelthomas2774 committed:
Fix reloading packed content

samuelthomas2774 committed:
Allow multiple users to use BetterDiscord on macOS and Linux

samuelthomas2774 committed:
Fix selecting user avatars in activity

samuelthomas2774 committed:
Fix reloading packed content

samuelthomas2774 committed:
Stop watching theme files when unloading themes

samuelthomas2774 committed:
Move updates to the correct location

samuelthomas2774 committed:
Remove duplicate collection styles

zerebos commented:
I like some of these bugfixes and the adjustments for macOS+Linux. Jiiks should be the one to review most of these things including packed reloading.

samuelthomas2774 committed:
Fix removing items from file settings

samuelthomas2774 committed:
Fix native open dialogs and attach the save packed content window as a modal

Maks-s committed:
Fix context menu console error

samuelthomas2774 committed:
Patch DOMTokenList methods to use the correct (not normalised) classes

rauenzi/BetterDiscordApp#159, rauenzi/BetterDiscordApp#161

samuelthomas2774 committed:
Discord Canary fixes

samuelthomas2774 committed:
Fix checking unknown components

Hopefully fixes the lag switching channels when Discord has been open for a while.

https://discordapp.com/channels/400734824740028417/423193907736870920/573535069764059157

samuelthomas2774 committed:
Comments and fix recording selectors for some components (and add one for VueComponent)

samuelthomas2774 committed:
Fix member list badges

Maks-s committed:
Fix ProfileBadge console error

samuelthomas2774 committed:
Fix member list profile badges

samuelthomas2774 committed:
Fix voice channel component filter

samuelthomas2774 committed:
Fix guild component selector
